lib/exe-type.js

```diff
@@ -0,0 +1,33 @@
+'use strict';
+
+const isWindows = require("is-windows")
+const path = require("path")
+
+function isCmd(file) {
+  const comspec = path.basename(process.env.comspec || '').replace(/\.exe$/i, '')
+  return isWindows() && (file === comspec || /^cmd(?:\.exe)?$/i.test(file))
+}
+
+function isNode(file) {
+  const cmdname = path.basename(process.execPath).replace(/\.exe$/i, '')
+  return file === 'node' || cmdname === file
+}
+
+function isNpm(file) {
+  // XXX is this even possible/necessary?
+  // wouldn't npm just be detected as a node shebang?
+  return file === 'npm' && !isWindows()
+}
+
+const KNOWN_SHELLS = new Set(['dash', 'sh', 'bash', 'zsh'])
+
+function isSh(file) {
+  return KNOWN_SHELLS.has(file)
```

Comment:
curious why this couldn't just be simplified to `['dash', 'etc'].includes(file)` since the `KNOWN_SHELLS` constant isn't reused anywhere

Reply:
Fixed